### ISSUE TITLE
Introduce pagination/infinite scrolling for the Samples page

### ISSUE THREAD

**Issue by qu8n** [mskcc/smile-server#1291]

**Comment by ao508** [comment]
Please [add your planning poker estimate](https://app.zenhub.com/workspaces/smile-scrum-5fcfa28a4ccc1600165347b6/issues/mskcc/smile-server/1291?planning-poker) with Zenhub @qu8n

**Comment by qu8n** [comment]
Quan & Angelica discussed the following: we'll be using the AG Grid's [Infinite Row Model](https://www.ag-grid.com/react-data-grid/infinite-scrolling/) for the Samples page's infinite scrolling instead of the SSRM version that non-Sample views currently use.

After trying the latter approach (SSRM), I found it not working well with our cell editing flow, especially around bulk editing (editing of multiple cells before submitting) and optimistic updates. I've exhausted the docs and online resources, and this topic -- SSRM cell editing -- is very under-explained. The majority of AG Grid's docs around cell editing focuses on other models.
